src/xrAICore/Components/problem_solver.h

```diff
@@ -151,4 +151,4 @@ class CProblemSolver
     virtual void clear();
 };
 
-#include "xrAICore/Components/problem_solver_inline.h"
+//#include "xrAICore/Components/problem_solver_inline.h"
```

Comment:
Are you sure?

Reply:
Yes. Check full commit - I moved it:
https://github.com/q4a/xray-16/commit/710cd10d41620c7b0047d4eb4ad3b03c0b68c68e

Reply:
The problem is the headers with '_inline' suffix in the name are usually included to the headers without the suffix (and only to this header). Moving the include could mislead somebody.

Reply:
We got some problems, when moved xrAICore from xrGame. One of them here - problem_solver_inline.h does not know about "ai()":
https://travis-ci.org/q4a/xray-16/jobs/462642365#L1952
When I added
#include "xrGame/ai_space.h"
I got another error:
xrAICore/Components/problem_solver_inline.h:367:36: error: member access into incomplete type 'CGraphEngine'
    m_failed = !ai().graph_engine().search(*this, reverse_search ? target_state() : current_state(),
                                                      ^
/home/1/code/xray-16/src/xrAICore/../xrAICore/AISpaceBase.hpp:8:7: note: forward declaration of 'CGraphEngine'

So CGraphEngine is main problem: it includes "xrAICore/Components/problem_solver.h" (L23) before adding template for "search"  (L104): 
https://github.com/OpenXRay/xray-16/blob/b4f778f72e1db6ef138094bd0a28685faacc316b/src/xrAICore/Navigation/graph_engine.h#L23

Reply:
@FreeZoneMods we should check this commit:
https://github.com/Zegeri/xray-16/commit/578e9423c132740b09f4381700adeaff9087b9c3
may be it will help to aviod removing problem_solver_inline.h  from problem_solver.h

Reply:
Maybe we should find a way to remove ai() call from this problem solver at all? Calling ai() itself seems strange here.
And seems like it's just a warning, not an error.

Reply:
I'll think about removing ai() from here, but I'm already checked Zegeri's commit: 
https://github.com/Zegeri/xray-16/commit/578e9423c132740b09f4381700adeaff9087b9c3
He fixed problem of ai(), but not fixed access into incomplete type 'CGraphEngine' - here is clang build log:
https://travis-ci.org/q4a/xray-16/jobs/469356988#L7898